src/chainstate/stacks/miner.rs

```diff
@@ -332,22 +568,46 @@ impl<'a> StacksMicroblockBuilder<'a> {
     }
 
     /// Mine the next transaction into a microblock.
-    /// Returns Some(StacksTransactionReceipt) or None if the transaction was
-    /// or was not mined into this microblock.
+    /// Returns Ok(TransactionResult::Success) if the transaction was mined into this microblock.
+    /// Returns Ok(TransactionResult::Skipped) if the transaction was not mined, but can be mined later.
+    /// Returns Ok(TransactionResult::Error) if the transaction was not mined due to an error.
```

Comment:
what is the difference between `Ok(TR:Error)` and `Err(e)`?

I.e., it's `transaction was not mined due to an error` vs. `if an error occurs during the function`.

I'm not understanding the dividing line from the comment text.

Reply:
The latter refers to an unexpected "operating" error in the function. An example of an operating error would be if a database query failed.